app/data_model/list_store.py

```diff
@@ -12,24 +13,80 @@ def random_string(length):
     return ''.join(random.choice(ascii_letters) for _ in range(length))
 
 
+class ListRepresentation:
+    def __init__(self, list_items: List = None, context: Mapping = None):
+        if not list_items:
+            list_items = list()
+        if not context:
+            context = dict()
+
+        self.list_items = list_items
+        self.context = context
+
+    def __eq__(self, other):
+        if not isinstance(other, ListRepresentation):
+            return NotImplemented
+
+        return self.list_items == other.list_items and self.context == other.context
+
+    def remove_item_id(self, to_remove):
+        self.list_items.remove(to_remove)
+
+    def __repr__(self):
+        return f'<ListRepresentation items={self.list_items}, context={self.context}>'
+
+
 class ListStore:
     """ Storage for lists.
+
+    Stores in the form:
+
+    ```
+    self._lists: {
+        list_name_1: ListRepresentation(
+            list_items: [1,2,3],
+            context: {'context_field': 'some_context'}
+        )
+    }
+    ```
+
+    serialises to:
+
+    ```
+    [
+        {
+            name: list_name_1,
+            context: {'context_field': 'some_context'}
+            items: [1,2,3]
+        }
+    ]
+    ```
     """
 
-    def __init__(self, existing_items=None):
+    def __init__(self, existing_items: Optional[List[Mapping]] = None):
         if not existing_items:
-            existing_items = []
+            existing_items = [{}]
+
+        self._lists: Mapping[str, ListRepresentation] = {
+            x.get('name'): ListRepresentation(
+                list_items=x.get('items'), context=x.get('context', {})
+            )
+            for x in existing_items
```

Comment:
`for existing_item in existing_items`? or just refer to them as list_items?

Reply:
I'm using `item` now, keeps it a little shorter

Reply:
Fixed.